Jellyfin.Server.Implementations/MediaSegments/MediaSegmentManager.cs

```diff
@@ -139,18 +139,47 @@ public async Task DeleteSegmentAsync(Guid segmentId)
     }
 
     /// <inheritdoc />
-    public async Task<IEnumerable<MediaSegmentDto>> GetSegmentsAsync(Guid itemId, IEnumerable<MediaSegmentType>? typeFilter)
+    public Task<IEnumerable<MediaSegmentDto>> GetSegmentsAsync(Guid itemId, IEnumerable<MediaSegmentType>? typeFilter, bool filterByProvider = true)
```

Comment:
David Fowler recommends using async-await over returning the Task directly: https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task

I don't think the slight performance gain (premature optimization) by not using async-await is worth mangling the stacktrace etc.

Reply:
I disagree that the performance difference is negaligable esp in our case where we usually use ConfigureAwait(false)

Reply:
Can you elaborate? I don't see how ConfigureAwait has any impact on whether we should use async-await or return tasks directly.